tests/common/database_fixture.cpp

```diff
@@ -208,16 +208,20 @@ database_fixture::database_fixture()
 }
 
 database_fixture::~database_fixture()
-{ try {
-   // If we're unwinding due to an exception, don't do any more checks.
-   // This way, boost test's last checkpoint tells us approximately where the error was.
-   if( !std::uncaught_exception() )
-   {
-      verify_asset_supplies(db);
-      BOOST_CHECK( db.get_node_properties().skip_flags == database::skip_nothing );
+{ 
+   try {
+      // If we're unwinding due to an exception, don't do any more checks.
+      // This way, boost test's last checkpoint tells us approximately where the error was.
+      if( !std::uncaught_exception() )
+      {
+         verify_asset_supplies(db);
+         BOOST_CHECK( db.get_node_properties().skip_flags == database::skip_nothing );
+      }
+      return;
+   } catch (fc::exception& ex) {
+      BOOST_FAIL( ex.to_detail_string() );
```

Comment:
Admittedly CAPTURE_AND_RETHROW seems pointless if it doesn't capture anything. Perhaps LOG_AND_RETHROW instead? Or why do you BOOST_FAIL on fc::exception only is better? (Note that fc macros also catch and handle other exceptions.)